modules/nvidia_plugin/src/cuda_config.cpp

```diff
@@ -10,71 +10,179 @@
 #include <error.hpp>
 #include <regex>
 
+#include "nvidia/properties.hpp"
+
 using namespace ov::nvidia_gpu;
 
 Configuration::Configuration() {}
 
+std::vector<ov::PropertyName> Configuration::get_ro_properties() {
+    static const std::vector<ov::PropertyName> ro_properties = {
+        // Metrics
+        ov::PropertyName{ov::available_devices.name(), ov::PropertyMutability::RO},
+        ov::PropertyName{ov::caching_properties.name(), ov::PropertyMutability::RO},
+        ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO},
+        ov::PropertyName{ov::range_for_streams.name(), ov::PropertyMutability::RO},
+        ov::PropertyName{ov::range_for_async_infer_requests.name(), ov::PropertyMutability::RO},
+        ov::PropertyName{ov::device::architecture.name(), ov::PropertyMutability::RO},
+        ov::PropertyName{ov::device::capabilities.name(), ov::PropertyMutability::RO},
+        ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO},
+        ov::PropertyName{ov::device::uuid.name(), ov::PropertyMutability::RO}
+    };
+    return ro_properties;
+}
+
+std::vector<ov::PropertyName> Configuration::get_rw_properties() {
+    static const std::vector<ov::PropertyName> rw_properties = {
+        // Configs
+        ov::PropertyName{ov::device::id.name(), ov::PropertyMutability::RW},
+        ov::PropertyName{ov::inference_precision.name(), ov::PropertyMutability::RW},
+        ov::PropertyName{ov::num_streams.name(), ov::PropertyMutability::RW},
+        ov::PropertyName{ov::hint::num_requests.name(), ov::PropertyMutability::RW},
+        ov::PropertyName{ov::hint::performance_mode.name(), ov::PropertyMutability::RW},
+        ov::PropertyName{ov::hint::execution_mode.name(), ov::PropertyMutability::RW},
+        ov::PropertyName{ov::enable_profiling.name(), ov::PropertyMutability::RW},
+        ov::PropertyName{ov::nvidia_gpu::operation_benchmark.name(), ov::PropertyMutability::RW},
+    };
+    return rw_properties;
+}
+
+bool Configuration::is_rw_property(const std::string& name) {
+    auto supported_rw_properties = get_rw_properties();
+    return (std::find(supported_rw_properties.begin(),
```

Comment:
This will work faster if use unordered set instead of vector

Reply:
> This will work faster if use unordered set instead of vector

This vector is also used for returning of supported properties. So it it better to fix order.